src/coreclr/System.Private.CoreLib/src/System/Reflection/RuntimePropertyInfo.cs

```diff
@@ -44,10 +44,64 @@ internal RuntimePropertyInfo(
 
             scope.GetPropertyProps(tkProperty, out m_utf8name, out m_flags, out _);
 
-            Associates.AssignAssociates(scope, tkProperty, declaredType, reflectedTypeCache.GetRuntimeType(),
+            RuntimeType reflectedRuntimeType = reflectedTypeCache.GetRuntimeType();
+            Associates.AssignAssociates(scope, tkProperty, declaredType, reflectedRuntimeType,
                 out _, out _, out _,
                 out m_getterMethod, out m_setterMethod, out m_otherMethod,
                 out isPrivate, out m_bindingFlags);
+
+            // lookup getter/setter when getter and setter are inherited from base class but just a setter/getter is overridden on a sub type
+            if ((m_getterMethod != null && m_getterMethod.IsVirtual && m_setterMethod == null)
+                || (m_setterMethod != null && m_setterMethod.IsVirtual && m_getterMethod == null))
+            {
+                Type? baseDeclaredType = declaredType.BaseType;
+
+                while ((m_getterMethod == null || m_setterMethod == null)
+                    && baseDeclaredType != null && baseDeclaredType is RuntimeType baseDeclaredRuntimeType)
```

Comment:
```suggestion
                    && baseDeclaredType is RuntimeType baseDeclaredRuntimeType)
```

The pattern match also does a null check.